libbeat/common/kubernetes/watcher.go

```diff
@@ -186,8 +194,14 @@ func (w *watcher) watch() {
 			if err != nil {
 				logp.Err("kubernetes: Watching API error %v", err)
 				watcher.Close()
+				if !(err == io.EOF || err == io.ErrUnexpectedEOF) {
+					// This is an error event which can be recovered by moving to the latest resource verison
+					logp.Info("kubernetes: Ignoring event, moving to most recent resource version")
+					w.lastResourceVersion = ""
```

Comment:
We may need to resync the stored objects, e.g. what would happen if we lose a delete event while we reconnect?

Reply:
I've been doing some tests. At least when I reproduce this last resource version is increased by several hundred while no real change happened.

I agree with you @jsoriano, a resync may be needed to guarantee no events are missed during the reconnect.

Still, taking into account the severity of this issue, I'm ok with merging + backporting this as it is, as it improves the current situation (infinite loop + crash). I can do some more research and come up with a resync mechanism afterwards.

What do you think?

Reply:
i think we are good to merge this one in.